### PR TITLE
Implement Mana Regeneration

### LINKED ISSUE #174: Add mana restoration
**User Story**
As a player, I'd like to be able recover my mana pool after having used some or all of it to cast spells.

**Acceptance Criteria**

- Mana potions added
- Passive mana regeneration added

**Notes**

---

**Why is this feature needed? Please describe the problem your requested feature wants to solve**
I am unable to recover mana that I have spent.

**Describe the solution you'd like**
I'd like to see mana potions added to the shop alongside the HP potions. I'd also like some sort of passive regeneration so that I don't have to rely fully on potions to restore my mana.

### PR DESCRIPTION
**Pull request by Pyxxil**
GitHub Issue (If applicable): #174 

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the current behavior?
The only way to get more mana after you have used it all is to level up.

## What is the new behavior?
Mana potions are available in the shop, and the player will also passively regain mana when out of combat.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current upstream repo
- [ ] Added new tests to the test suite for the feature
- [ ] Tested the full test suite, and ensured that the feature does not break current build.
- [ ] Docs have been added/updated which fit (for bug fixes / features)
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal)
- [ ] Checked licensing of Frameworks/Libraries (if applicable) 

## Other information
This isn't set in stone -- currently the player has to be out of combat for 10 turns (and every 10 out of combat turns thereafter) to regen 1 mana passively. I think this really should instead be relative to the amount they would regenerate with a potion (not entirely sure the amount at the moment).

Internal Issue (If applicable): closes #174

### CODE REVIEW

**Comment by Pyxxil** [comment]
This should be merged after #181, so that we can avoid any issues regarding some refactoring.

File: src/data/items/potions/great-hp-potion/index.js [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import GreatHpPotionImg from './great-hp-potion.png';
 
 const GreatHpPotion = {
-    name: 'Great HP Potion',
+    name: 'Great Health Potion',
     type: 'potion',
+    kind: 'mana',
```
Comment (Pyxxil) [inline src/data/items/potions/great-hp-potion/index.js]: This should probably be 'health'

File: src/components/view-item/index.js [hunk 2]
```diff
@@ -395,7 +396,7 @@ const ViewItem = ({
                 open={confirmPotion}
                 text={`Are you sure you want to use your ${data.name}?`}
                 cancelText={'Cancel'}
-                acceptText={'Heal'}
+                acceptText={data.kind == 'health' ? 'Heal' : 'Restore'}
```
Comment (Matteas-Eden) [inline src/components/view-item/index.js]: Should be `===` instead of `==`.

**Review by Matteas-Eden — CHANGES_REQUESTED** [review]
There don't seem to be any mana potions available until level 10. Could you add `Mana Potion` and `Greater Mana Potion` to the first shop tier?

**Review by Matteas-Eden — CHANGES_REQUESTED** [review]
In addition to the conflict with `master` that needs to be resolved, I also get an issue with having a defence value of `NaN`. Perhaps that won't persist if this gets merged, but I'd like some confirmation that it won't produce an issue before I approve.

**Comment by Pyxxil** [comment]
> In addition to the conflict with `master` that needs to be resolved, I also get an issue with having a defence value of `NaN`. Perhaps that won't persist if this gets merged, but I'd like some confirmation that it won't produce an issue before I approve.

I can't reproduce this, and even if so, it wasn't introduced as an issue by this PR and probably should have been caught in a seperate issue.

**Review by Matteas-Eden — APPROVED** [review]
The previous error I mentioned was a run-time error due to my own testing setup, so apologies for that. LGTM.

**Pull request merged**